DataFormats/L1TParticleFlow/interface/HPSPFTau.h

```diff
@@ -0,0 +1,154 @@
+    float rhoCorr() const { return rhoCorr_; }
+
+    bool passTightIso() const { return passTightIso_; }
+    bool passMediumIso() const { return passMediumIso_; }
+    bool passLooseIso() const { return passLooseIso_; }
+    bool passVLooseIso() const { return passVLooseIso_; }
+
+    bool passTightRelIso() const { return passTightRelIso_; }
+    bool passMediumRelIso() const { return passMediumRelIso_; }
+    bool passLooseRelIso() const { return passLooseRelIso_; }
+    bool passVLooseRelIso() const { return passVLooseRelIso_; }
+
+    friend class ::HPSPFTauBuilder;
+
+  private:
+    l1t::PFCandidateRef seedChargedPFCand_;
+    l1t::PFJetRef seedPFJet_;
+    l1t::PFCandidateRef leadChargedPFCand_;
+
+    l1t::PFCandidateRefVector signalAllL1PFCandidates_;
+    l1t::PFCandidateRefVector signalChargedHadrons_;
+    l1t::PFCandidateRefVector signalElectrons_;
+    l1t::PFCandidateRefVector signalNeutralHadrons_;
+    l1t::PFCandidateRefVector signalPhotons_;
+    l1t::PFCandidateRefVector signalMuons_;
+
+    l1t::PFCandidateRefVector stripAllL1PFCandidates_;
+    l1t::PFCandidateRefVector stripElectrons_;
+    l1t::PFCandidateRefVector stripPhotons_;
+
+    l1t::PFCandidateRefVector isoAllL1PFCandidates_;
+    l1t::PFCandidateRefVector isoChargedHadrons_;
+    l1t::PFCandidateRefVector isoElectrons_;
+    l1t::PFCandidateRefVector isoNeutralHadrons_;
+    l1t::PFCandidateRefVector isoPhotons_;
+    l1t::PFCandidateRefVector isoMuons_;
+
+    l1t::PFCandidateRefVector sumAllL1PFCandidates_;
+    l1t::PFCandidateRefVector sumChargedHadrons_;
+    l1t::PFCandidateRefVector sumElectrons_;
+    l1t::PFCandidateRefVector sumNeutralHadrons_;
+    l1t::PFCandidateRefVector sumPhotons_;
+    l1t::PFCandidateRefVector sumMuons_;
+
+    //l1t::VertexRef primaryVertex_;
+    l1t::TkPrimaryVertexRef primaryVertex_;
+    Kind tauType_;
+
+    reco::Particle::LorentzVector strip_p4_;
+
+    float sumAllL1PFCandidates_pt_;
+    float signalConeSize_;
+    float isolationConeSize_;
+
+    float sumChargedIso_;
+    float sumNeutralIso_;
+    float sumCombinedIso_;
+    float sumChargedIsoPileup_;  // charged PFCands failing dz cut (maybe useful to correct neutral isolation for pile-up contributions by applying delta-beta corrections)
+    float rhoCorr_;              // rho correction (maybe useful for applying pile-up corrections to neutral isolation)
+
+    bool passTightIso_;
+    bool passMediumIso_;
+    bool passLooseIso_;
+    bool passVLooseIso_;
+
+    bool passTightRelIso_;
+    bool passMediumRelIso_;
+    bool passLooseRelIso_;
+    bool passVLooseRelIso_;
+  };
+
+}  // namespace l1t
+
+/// print to stream
+std::ostream& operator<<(std::ostream& os, const l1t::HPSPFTau& l1PFTau);
+
+//void printPFCand(ostream& os, const l1t::PFCandidate& l1PFCand, const l1t::VertexRef& primaryVertex);
+void printPFCand(ostream& os, const l1t::PFCandidate& l1PFCand, const l1t::TkPrimaryVertexRef& primaryVertex);
```

Comment:
maybe these functions should be in the `l1t` namespace (rather than the global namespace)?

Reply:
Done!